src/math/random.js

```diff
@@ -131,13 +131,12 @@ p5.prototype.random = function (min, max) {
   } else {
     rand = Math.random();
   }
-
-  if (arguments.length === 0) {
+  if (typeof min === 'undefined') {
     return rand;
   } else
-  if (arguments.length === 1) {
-    if (arguments[0] instanceof Array) {
-      return arguments[0][Math.floor(rand * arguments[0].length)];
+  if (typeof max === 'undefined') {
```

Comment:
I agree, in this case I think we can prioritize efficiency. Generally, I think we want to be taking measures to catch common errors, but don't need to safeguard against every possible one.